dev-packages/eslint-plugin/index.js

```diff
@@ -17,5 +17,6 @@
 
 /** @type {{[ruleId: string]: import('eslint').Rule.RuleModule}} */
 exports.rules = {
-    "shared-dependencies": require('./rules/shared-dependencies'),
+    "runtime-import-check": require('./rules/runtime-import-check'),
```

Comment:
This name could be a little bit misleading, since it could sound like a check on imports made at runtime (I'm reminded of Paul's recent PR re: dynamic `require`). Maybe something like `code-partition`, `code-partition-check` or `code-partition-integrity`?

Reply:
No no, it really depends on the runtime in the sense of Node runtime? Browser runtime? Electron runtime? Runtime means execution engine/environment? WDYT?

Reply:
Yes, I understand what sense of runtime is intended, but I think it might be better to choose a name without the ambiguity of that term.

Reply:
@colin-grant-work I'm not really found of the use of `partition` here, I think it only adds additional confusion (it's not intuitive for me, and I can imagine for downstream and newcomers it won't be either). Originally I had `code-organization`, but agreed with Paul that we are really verifying that we are importing correctly for the different runtimes we support.

Perhaps it'd be clearer like:

```
check-import-for-runtimes
```

or the more generic rule name:

```
no-invalid-import
```

Reply:
Yes, I liked the `code-organization` name better :-). If there are more votes for `runtime` or some variation, I won't press too insistently. Maybe expand the message to something like "<Module> cannot be imported into <matched folder> to avoid incompatibility with target runtime." ?